aderyn_core/src/detect/high/high_level_calls_in_loop.rs

```diff
@@ -0,0 +1,195 @@
+use std::collections::BTreeMap;
+use std::convert::identity;
+use std::error::Error;
+
+use crate::ast::{ASTNode, Expression, MemberAccess, NodeID};
+
+use crate::capture;
+use crate::context::browser::ExtractFunctionCalls;
+use crate::context::investigator::{
+    StandardInvestigationStyle, StandardInvestigator, StandardInvestigatorVisitor,
+};
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct HighLevelCallsInLoopDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+/**
+    This [`HighLevelCallsInLoopDetector`] detector will catch address.transfer as well as high level calls in loops
+    (as they may potentially revert) and we don't specify "revert" explicitly.
+    1. address.transfer
+    2. Contract.xyz()
+
+    Things what we don't capture - `address.send`, `address.call`, etc
+    (Because we know for sure they don't revert unless we ask them to based on the return value of bool success)
+
+    The solution is to perform this function call with low level
+    call and then you can just break off the loop if the bool success returns false!
+
+    In order to catch `requires` and `reverts` in the loops, there is another detector we have in `reverts_and_requires_in_loop.rs`
+*/
```

Comment:
@alexroan It's explained here. 

But I'll change the description to match slither's :)

Reply:
TBH I'm unsure of this detector and this was the one for which you said you'll have a think about whether or not to consider because pull over  push is more of an archaic concept now

Reply:
I'm totally okay with not proceeding with this. What do you say?

Reply:
I'm thinking about it, I'm not ready to merge this yet based on this discussion.

Can you please:
1. Link this PR on the ticket
2. Move the ticket back to the backlog
3. Close this PR

We can reassess whether we want to tackle this one at the sync meeting.

Reply:
Yessir!